src/hooks/resources/service_instance.ts

```diff
@@ -8,7 +8,7 @@ import { getIP } from '../../lib/utils';
 sbvrUtils.addPureHook('POST', 'resin', 'service_instance', {
 	POSTPARSE: ({ request, req }) => {
 		request.values.service_type = getServiceFromRequest(req);
-		request.values.ip_address = getIP(req as Request);
+		request.values.ip_address = req?.body?.ip_address ?? getIP(req as Request);
```

Comment:
I'm pretty sure most of these `?` are unnecessary, at a minimum
```suggestion
		request.values.ip_address = req.body?.ip_address ?? getIP(req as Request);
```
but I think
```suggestion
		request.values.ip_address = req.body.ip_address ?? getIP(req as Request);
```
would also work

Reply:
Yeah, you're right… I think I got a bit carried away ha.